### PR TITLE
cli: properly load specified method for `run` VM CLI command

### PR DESCRIPTION
**Pull request by AnnaShaleva**
Ref. #3013, based on https://github.com/nspcc-dev/neo-go/pull/3013#issuecomment-1545949629.

TODO:
- [x] Fix currently failing tests.
- [x] Write more tests.

### CODE REVIEW

File: cli/vm/cli.go [hunk 1]
```diff
@@ -828,20 +857,32 @@ func handleLoadGo(c *cli.Context) error {
 		}
 		if len(args) < 3 {
 			return fmt.Errorf("%w: signers expected after `%s`, got none", ErrInvalidParameter, cmdargs.CosignersSeparator)
+
```
Comment (roman-khimov) [inline cli/vm/cli.go:860]: :eyes:

File: cli/vm/cli.go [hunk 2]
```diff
@@ -828,20 +857,32 @@ func handleLoadGo(c *cli.Context) error {
...
+		Hash:     h,
+		NEF:      *ne,
+		Manifest: *m,
+	}
+	setContractStateInContext(c.App, cs)
+
 	v := getVMFromContext(c.App)
-	setManifestInContext(c.App, m)
 	fmt.Fprintf(c.App.Writer, "READY: loaded %d instructions\n", v.Context().LenInstr())
+
```
Comment (roman-khimov) [inline cli/vm/cli.go:885]: :eyes:

File: cli/vm/cli.go [hunk 3]
```diff
@@ -1045,27 +1085,35 @@ func handleRun(c *cli.Context) error {
...
+			// To handle it properly, we need to unload previously loaded context (and it is done). But then
+			// we have to manually reset the gas limit to the previous value and (which is more important)
+			// preserve the same set of breakpoints, see TestRunWithBreakpoint.
+			// Are you sure that we really want to use v.LoadNEFMethod here and change the executable context?
+
+			ic := getInteropContextFromContext(c.App)
+			gasLimit := v.GasLimit
+			ic.ReuseVM(v)         // Clear context loaded by 'loadgo', 'loadnef' or 'loaddeployed' to properly handle LoadNEFMethod.
+			v.GasLimit = gasLimit // Preserve the gas limit set by loading commands.
+			v.LoadNEFMethod(&cs.NEF, util.Uint160{}, cs.Hash, callflag.All, hasRet, offset, initOff, nil)
```
Comment (AnnaShaleva) [inline cli/vm/cli.go:1112]: @roman-khimov, could you take a look, please?

Comment (roman-khimov) [reply cli/vm/cli.go:1112]: What are the alternatives? LoadNEFMethod is just a (known, correct) way to set things up for a contract. IIRC there was some problem with passing NEF itself into the context for it to work correctly (for example `CALLT`).

Comment (AnnaShaleva) [reply cli/vm/cli.go]: Fixed.

File: cli/vm/cli.go [hunk 4]
```diff
@@ -489,7 +496,7 @@ func NewWithConfig(printLogotype bool, onExit func(int), c *readline.Config, cfg
 		chainKey:            chain,
 		chainCfgKey:         cfg,
 		icKey:               ic,
-		manifestKey:         new(manifest.Manifest),
+		contractStateKey:    new(manifest.Manifest),
```
[the remark this answers is not in the document]
Comment (AnnaShaleva) [reply cli/vm/cli.go]: Fixed, I've missed this.

**Review by roman-khimov — APPROVED** [review]

**Pull request merged**